tools/sdk/raydium/helpers.ts

```diff
@@ -32,16 +32,53 @@ export const getAmountOut = async (
       poolKeys,
     }),
     currencyAmountIn: new TokenAmount(
-      new Token(new PublicKey(tokenInData.address), tokenInData.decimals),
+      new Token(poolKeys.baseMint, base.value.decimals),
       amountInBN
     ),
-    currencyOut: new Token(
-      new PublicKey(tokenOutData.address),
-      tokenOutData.decimals
-    ),
+    currencyOut: new Token(poolKeys.quoteMint, quote.value.decimals),
     slippage: new Percent(5, 1000),
   })
-  const currentPrice = amountOut.currentPrice.toFixed(tokenOutData.decimals)
+  const currentPrice = amountOut.currentPrice.toFixed(quote.value.decimals)
 
   return Number(currentPrice) * amountIn
 }
+
+export const getLPMintInfo = async (
+  connection: ConnectionContext,
+  lpMint: PublicKey,
+  user: PublicKey
+) => {
+  const [lpTokenAccount] = findATAAddrSync(user, lpMint)
+  const lpInfo = await connection.current.getTokenSupply(lpMint)
+  const lpUserBalance = await connection.current.getTokenAccountBalance(
+    lpTokenAccount
+  )
+  return {
+    lpTokenAccount,
+    maxBalance: lpUserBalance.value.uiAmount ?? 0,
```

Comment:
if there is no amount, do we really want the `maxBalance` to be 0?

Reply:
I thinj it would throw if there is no amount, the 0 value is to cover in case it still goes through somehow. If you have any suggestion, I'm up to it

Reply:
yeah you may throw :)